### DISCUSSION TITLE
Fix CVE-2022-27204 / SECURITY-1350

### DISCUSSION THREAD

**Discussion by chonton** [jenkinsci/extended-choice-parameter-plugin#55] [General]
[CSRF vulnerability and missing permission checks allow SSRF](https://www.jenkins.io/security/advisory/2022-03-15/#SECURITY-1350) security advisory mentions both permission checks and CSRF.  CRSF was done in #39.  @daniel-beck What's left to complete with permission checks?

**Comment by daniel-beck** [comment]
Unsure what the question is. Are you looking for https://www.jenkins.io/doc/developer/security/form-validation/ ?

**Reply by chonton** [reply]
[CVE-2022-27205](https://cve.mitre.org/cgi-bin/cvename.cgi?name=2022-27205) states "A missing permission check ... allows attackers with Overall/Read permission to connect to an attacker-specified URL".  I don't understand how _any_ permission check could mitigate this concern.  If the ability to use urls is removed, (lines [134-136](https://github.com/jenkinsci/extended-choice-parameter-plugin/blob/master/src/main/java/com/cwctravel/hudson/plugins/extended_choice_parameter/ExtendedChoiceParameterDefinition.java#L135)), can the security advisory be closed? 

(This capability to use any url for properties is not documented.)

Additionally, since [doCheckPropertyFile](https://github.com/jenkinsci/extended-choice-parameter-plugin/blob/master/src/main/java/com/cwctravel/hudson/plugins/extended_choice_parameter/ExtendedChoiceParameterDefinition.java#L119) does not implement additional permission checks, this method is a golden oracle for whether a particular properties file exists and whether a particular property exists within that file.  Is this information leakage sufficient to keep the security advisory open?
